From 72cfa6ac1ce9de5bde61414bf874ce19057df9d3 Mon Sep 17 00:00:00 2001
From: wuzhouhui <wuzhouhui250@gmail.com>
Date: Tue, 9 Jan 2018 22:22:58 +0800
Subject: [PATCH 2/2] Do not set properties_only if item deleted is a dir

If we set properties_only for deleted directory and there is files
in it, then there is not diff output of these files in log message,
this not what we want!
---
 subversion/libsvn_client/diff.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/subversion/libsvn_client/diff.c b/subversion/libsvn_client/diff.c
index 7020c02..fe9ebfb 100644
--- a/subversion/libsvn_client/diff.c
+++ b/subversion/libsvn_client/diff.c
@@ -2912,7 +2912,10 @@ verbose_output(const apr_array_header_t *commit_items,
                TRUE,                          /* show_copies_as_adds */
                FALSE,                         /* ignore_content_type */
                FALSE,                         /* ignore_properties */
-               item->kind == svn_node_dir,    /* properties_only */
+               /* properties_only */
+               !(item->kind != svn_node_dir
+                 || (item->kind == svn_node_dir
+                     && (item->state_flags & SVN_CLIENT_COMMIT_ITEM_DELETE))),
                FALSE,                         /* use_git_diff_format */
                svn_cmdline_output_encoding(pool),
                fstream,
-- 
2.14.1

